### PR TITLE
Warn on exit

### PR DESCRIPTION
**Pull request by ztolley**
Most browsers warn the user that they are leaving a page and have made changes to a form. This small polyfill listens for form changes and adds an event to warn the user if they try to leave the page, unless they submit the form in which case it removes the event handler that does that.

The reason the event is added and taken away is that if we just try to return false or something from that event handler it puts that on the screen. The easiest way to handle things is remove the even handler altogether when you submit a page as it's no longer needed.

### CODE REVIEW

File: app/javascript/components/highlightcontrol.js [hunk 1]
```diff
@@ -29,9 +29,11 @@ class HighlightControl {
...
 
   static init (container = document) {
-    container.querySelectorAll(`.${wrapperClass}`).forEach((element) => {
-      new HighlightControl(element)
-    })
+    Array
+      .from(container.querySelectorAll(`.${wrapperClass}`))
+      .forEach((element) => {
+        new HighlightControl(element)
+      })
```
Comment (rgarner) [inline app/javascript/components/highlightcontrol.js]: Is this related to warn on exit at all?

**Comment by rgarner** [comment]
The comment from the PR "The reason the event is added and taken away is that if we just try to return false or something from that event handler it puts that on the screen. The easiest way to handle things is remove the even handler altogether when you submit a page as it's no longer needed." is useful intel we don't want to lose. Can we add it to the commit message?

File: app/javascript/components/highlightcontrol.js (hunk 1 above)
Comment (ztolley) [inline app/javascript/components/highlightcontrol.js]: No, it's a bug I found whilst I was running the code on ie
